src/Frame/Frame.cc

```diff
@@ -26,6 +26,18 @@
 #include "ImageStats/StatsCalculator.h"
 #include "Logger/Logger.h"
 
+#if defined(__APPLE__) && defined(__MACH__)
+#include <Availability.h>
+#endif
+
+#if defined(__APPLE__) && defined(__MACH__) && defined(__MAC_OS_X_VERSION_MIN_REQUIRED)
+#if __MAC_OS_X_VERSION_MIN_REQUIRED < 110000
+// nothing to define here
+#else
+#define HAS_SHARED_PTR_ARRAY
```

Comment:
Surely this should *also* be defined if you're not on a mac at all? And only *not* defined if you're on a mac and it's old? This leaves it undefined if you're not on a mac (which was not the case in the earlier code).

This is clearer if you collapse it into a single conditional (please check my negation logic):

```
#if defined(__APPLE__) && defined(__MACH__) && defined(__MAC_OS_X_VERSION_MIN_REQUIRED) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 110000
// nothing to define here
#else
#define HAS_SHARED_PTR_ARRAY
#endif
```